### ISSUE TITLE
How do I effectively manage/control the links between states in state-diagram?

### ISSUE THREAD

**Issue by anandakumarpalanisamy** [mermaid-js/mermaid#3430]
## How do I effectively manage/control the links between states in state-diagram? I understand it could tough to get a diagram where links between states doesn't overlap. But is there a way that I can better manage.? Any help much appreciated.

[Link for live editor](https://mermaid.live/edit#pako:eNpd0bEOgjAQBuBXaW40-gIMJlQgYaIJbNahoVUbBUwtgyG8u8idScvGx_392-QmaAdtIIG3V95kVt2c6mTPmLbOtN4OPWv4z4yddxd2OBxZikxX0IyvqBH1igpRhRNEFYkaqjosJ_HwKkGNgir5hiW6WZnRU3isMmSGtfGsiQ6SBCYpKpCnmGkYPYl8k40tiqj4ryJqQgnYQ2dcp6xe1jT9ZhL83XRGQrJ8auUeEmQ_L7nxpZct5tr6wUHi3Wj2oEY_1J--heSqnm_zD9Gq6e_8BWtYgkc)

Markdown:

stateDiagram
  direction TB
    [*] --> A
    A --> B
    B --> S
    S --> O
    O --> S
    O --> OS
    O --> B
    OS --> A
    OS --> B
    A --> PO
    PO --> SB
    PO --> SBI
    T --> D
    SB --> D
    SBI --> D
    D --> P
    SBI --> T
    SB --> T
    P --> PP
    PP --> CP
    PP --> A
    P --> CPE
    PP --> CPE
    PP --> PF
    P --> PF
    PF --> A
    PF --> P

<img width="514" alt="image" src="https://user-images.githubusercontent.com/20585650/188923211-2edf6b01-408e-4c42-8881-aa6033d32dfc.png">

**Comment by weedySeaDragon** [comment]
Links are going to be drawn in the order that you define them,  and in a diagram going from Top to Bottom (direction = `TB`), when you have many links _(relations)_ from the same node _(state)_, the first link be drawn first on the left and whatever node (state) it ends on will be drawn. Then  the next link is drawn to the right of that first one and the node it ends on will be drawn and probably also be to the right of that previous node and so on.  

So if you have links that cross, see if you can change the order of your statements so that links (and the nodes they end on) are moved to the right or left -- whatever you need to fix the crossing lines.

For example with this section of your diagram:
```mermaid
stateDiagram
direction TB
[*] --> A
A --> B
B --> S
S --> O
O --> S
O --> OS
O --> B
OS --> A
OS --> B

```

You can see that the link OS --> B is crossing OS --> A.  But if you can get OS --> B to be drawn _to the left_ of OS --> A, you can fix it.  
In other words, you want OS --> B to be drawn before OS --> A.  So just change the order of your statements:

| original order of the statments (crossing lines) | revised order -- no crossing lines |
|-----------------------------------------------|-------------------|
| `OS --> A` |   `OS --> B` |
| `OS --> B` | `OS --> A` |

With that simple re-ordering of those statements, you get this:

```mermaid
stateDiagram
direction TB
[*] --> A
A --> B
B --> S
S --> O
O --> S
O --> OS
O --> B
OS --> B
OS --> A
```

That's a really good question.  I don't think that's clear anywhere in the documentation; we should make sure to explain it somwhere.
